torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1362,6 +1362,265 @@ def rbinary_magic_impl(self, other):
 del method
 del func
 
+try:
+    import z3
```

Comment:
nit. We can move all this to its own module and check that `z3` exists at the top of the module, rather than having a huge chunk of code inlined and indented here. Also, I'm sure that this code will be useful in other places later down the road.

Reply:
Maybe I could do it in a follow up PR. What do you think?